File: test_websocket_connection.js
// Simple WebSocket test using Node.js
const WebSocket = require('ws');

async function testWebSocket() {
    const wsUrl = 'ws://localhost:8000/ws/E8BF435B/admin';

    console.log(`Verbinde mit ${wsUrl}...`);

    const ws = new WebSocket(wsUrl);

    ws.on('open', function open() {
        console.log('✅ WebSocket-Verbindung erfolgreich!');

        // Sende Test-Nachricht
        const testMessage = {
            type: 'text_message',
            content: 'Hello WebSocket Test',
            timestamp: new Date().toISOString()
        };

        ws.send(JSON.stringify(testMessage));
        console.log('📤 Test-Nachricht gesendet');

        // Halte Verbindung 3 Sekunden offen
        setTimeout(() => {
            ws.close();
            console.log('🔌 WebSocket-Verbindung geschlossen');
        }, 3000);
    });

    ws.on('message', function message(data) {
        console.log(`📥 Antwort erhalten: ${data}`);
    });

    ws.on('error', function error(err) {
        console.error(`❌ WebSocket Fehler: ${err.message}`);
    });

    ws.on('close', function close() {
        console.log('🔌 Verbindung beendet');
        process.exit(0);
    });
}

testWebSocket();

File: scripts/test_websocket_connection.py
#!/usr/bin/env python3
import asyncio
import websockets
import json

async def test_websocket():
    """Test WebSocket connection to trigger metrics collection"""
    uri = "ws://localhost:8000/ws/2B73C3C3?client_type=customer"

    try:
        print(f"Verbinde mit {uri}...")
        async with websockets.connect(uri) as websocket:
            print("✅ WebSocket-Verbindung erfolgreich!")

            # Sende eine Test-Nachricht
            test_message = {
                "type": "text_message",
                "content": "Hello WebSocket Test",
                "timestamp": "2025-11-05T10:00:00Z"
            }

            await websocket.send(json.dumps(test_message))
            print("📤 Test-Nachricht gesendet")

            # Warte auf Antwort
            try:
                response = await asyncio.wait_for(websocket.recv(), timeout=5.0)
                print(f"📥 Antwort erhalten: {response}")
            except asyncio.TimeoutError:
                print("⏰ Timeout - keine Antwort erhalten")

            # Halte Verbindung kurz offen um Metriken zu generieren
            await asyncio.sleep(2)

        print("🔌 WebSocket-Verbindung geschlossen")

    except Exception as e:
        print(f"❌ Fehler: {e}")

if __name__ == "__main__":
    asyncio.run(test_websocket())